pypeit/spectrographs/subaru_focas.py

```diff
@@ -0,0 +1,688 @@
+            adjusted for configuration specific parameter values.
+        """
+        # Start with instrument wide
+        par = super().config_specific_par(scifile, inp_par=inp_par)
+
+        if self.get_meta_value(scifile, 'dispname') == 'SCFCGRMB01':
+            par['calibrations']['wavelengths']['reid_arxiv'] = 'wvarxiv_subaru_focas_SCFCGRMB01.fits'
+            par['calibrations']['wavelengths']['method'] = 'full_template'
+            par['calibrations']['wavelengths']['stretch_func'] = 'quadratic'
+        # ---- NOTE: from Debora ----
+        # The pypeit_sensfunc script uses the config_specific_par() method
+        # with a reduced spec1d file to pull out some info, although the method
+        # is meant for raw frames. In this case, the spec1d file does not have
+        # the dispname meta value in the header and PypeIt tries to run those
+        # 2 lines of code (just a message to the terminal) and crashes.
+        # So, removing them should fix the problem.
+        # ----------------------------
+        # else:
+        #     msgs.error(f'Not ready for this grism {self.get_meta_value(scifile, "dispname")}')
+
+        return par
+
+    def config_independent_frames(self):
+        """
+        Define frame types that are independent of the fully defined
+        instrument configuration.
+
+        This method returns a dictionary where the keys of the dictionary are
+        the list of configuration-independent frame types. The value of each
+        dictionary element can be set to one or more metadata keys that can
+        be used to assign each frame type to a given configuration group. See
+        :func:`~pypeit.metadata.PypeItMetaData.set_configurations` and how it
+        interprets the dictionary values, which can be None.
+
+        Returns:
+            :obj:`dict`: Dictionary where the keys are the frame types that
+            are configuration-independent and the values are the metadata
+            keywords that can be used to assign the frames to a configuration
+            group.
+        """
+        return {'bias': 'detector', 'dark': 'detector'}
+
+    def configuration_keys(self):
+        """
+        Return the metadata keys that define a unique instrument
+        configuration.
+
+        This list is used by :class:`~pypeit.metadata.PypeItMetaData` to
+        identify the unique configurations among the list of frames read
+        for a given reduction.
+
+        Returns:
+            :obj:`list`: List of keywords of data pulled from file headers
+            and used to constuct the :class:`~pypeit.metadata.PypeItMetaData`
+            object.
+        """
+        #return ['dispname', 'dispangle', 'decker', 'detector']
+        # TODO -- Consider dispangle
+        return ['dispname', 'decker', 'detector']
+
+
+    # TODO -- Convert this into get_comb_group()
+    def parse_dither_pattern(self, file_list, ext=None):
```

Comment:
do we actually use this?  if not, please remove.

Reply:
@Tang-SL , did you add this? Pretty sure our team did not, as we were not addressing dithers.

Reply:
@ejeschke Nope, that wasn't me.

Reply:
Great, thanks for chiming in.  I'll remove it then.

Reply:
Removed